File: competitive_programming/leetcode/01_python/9_palindrome_number.py
"""LeetCode 9 - Palindrome Number."""
def palindrome_number(x):
    return x >= 0 and str(x) == str(x)[::-1]


if __name__ == "__main__":
    assert palindrome_number(121) is True
    assert palindrome_number(-121) is False
    assert palindrome_number(10) is False
    print("9 palindrome number ok")

File: competitive_programming/leetcode/03_cpp/9_palindrome_number.cpp
#include <algorithm>
#include <cassert>
#include <cmath>
#include <cstdint>
#include <iostream>
#include <set>
#include <string>
#include <vector>

bool palindromeNumber(int x) {
    if (x < 0) return false;
    std::string digits = std::to_string(x);
    return std::equal(digits.begin(), digits.begin() + digits.size() / 2, digits.rbegin());
}

int main() {
    assert(palindromeNumber(121) && !palindromeNumber(-121) && !palindromeNumber(10));
    std::cout << "9 palindrome number ok\n";
    return 0;
}
